Lab 2 Changes (#252)
aws-samples/aws-hpc-tutorials: content/04-container-parallelcluster/02-update-PC.md

+++
title = "b. Update your cluster"
date = 2019-09-18T10:46:30-04:00
weight = 30
tags = ["tutorial", "update", "ParallelCluster"]
+++

In this section, you will update the configuration of the HPC cluster you created in [Lab I](03-hpc-aws-parallelcluster-workshop.html) to:
- Add a script to install Docker and Singularity.
- Provide access to the container registry, [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/).
- Create a new queue that will be used to run the containerized workload.
- Update the configuration of the HPC Cluster.

#### 1. Edit Cluster

Click on the **Edit** button in Pcluster Manager.

![Edit button](/images/container-pc/edit.png)

#### 2. HeadNode

> The first screen we're going to leave as is, next advance to the **HeadNode** tab.

Reviewer: ```suggestion
The first screen you are going to leave as is, next advance to the **HeadNode** tab.
```

> On the HeadNode tab we're going to add permission to access the [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) by adding the managed `AmazonEC2ContainerRegistryFullAccess` [AWS IAM](https://aws.amazon.com/iam/) policy.

Reviewer: ```suggestion
On the HeadNode tab you are going to add permission to access the [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) by adding the managed `AmazonEC2ContainerRegistryFullAccess` [AWS IAM](https://aws.amazon.com/iam/) policy.
```

1. Click the drop down on the **Advanced options**
2. Click the drop down on **IAM Policies**
3. Add in the policy `arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryFullAccess`. Click Add.

![HeadNode IAM](/images/container-pc/headnode-iam.png)

#### 3. Queue Configuration

> Click next twice to advance to the **Queues section**, here we're going to add a queue that has Docker and Singularity installed on the compute nodes. 

Reviewer: ```suggestion
Click next twice to advance to the **Queues section**, here you are going to add a queue that has Docker and Singularity installed on the compute nodes. 
```

> 1. Click **Add Queue**

Reviewer: ```suggestion
1. Choose **Add Queue**
```

2. Set the **Subnet** to the same subnet as the first queue (queue1)
3. Set the **Dynamic Nodes** to `8`
4. Set the **Instance Type** to `c5.xlarge`

![Queue Edit](/images/container-pc/queue-edit.png)

> Next we're going to add in a script that installs Docker and Singularity on the Compute Nodes.

Reviewer: ```suggestion
Next, you are going to add in a script that installs Docker and Singularity on the Compute Nodes.
```

1. Dropdown **Advanced Options** on the queue you just created
2. Paste in the following url into **On Configured** section `https://github.com/aws-samples/aws-hpc-tutorials/blob/isc22/static/scripts/post-install/container-install.sh`
3. Expand **IAM Policies** and paste in the following policy `arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryFullAccess`. Click Add.

![Advanced Options](/images/container-pc/queue-iam.png)

> #### 4. Increase RootVolume Size of your cluster

Reviewer: It is unclear where this change needs to be done.

> In the cluster's config we need to add the following snippet at the bottom of the `queue1` section, `line 56`.

Reviewer: ```suggestion
In the cluster's configuration, you need to add the following snippet at the bottom of the `queue1` section, `line 56`.
```

```yaml
ComputeSettings:
  LocalStorage:
    RootVolume:
      Size: 50
```

#### 5. Update your HPC Cluster

> On the next screen we'll confirm the cluster configuration and update the cluster.

Reviewer: ```suggestion
On the next screen, you will confirm the cluster configuration and update the cluster.
```

> 1. Click **Stop Compute Fleet** and click to confirm, this will take a minute to complete, wait to run the update until it's stopped.

Reviewer: ```suggestion
1. Choose **Stop Compute Fleet** and click to confirm, this will take a minute to complete, wait to run the update until it's stopped.
```

> 2. **Dryrun** to validate the cluster configuration. You'll see three warnings that you can safely ignore.

Reviewer: ```suggestion
3. **Dryrun** to validate the cluster configuration. You'll see three warnings that you can safely ignore.
```

> 3. Run **Update**

Reviewer: ```suggestion
4. Run **Update**
```

![Update Cluster](/images/container-pc/update-cluster.png)

> Once we've ran the update we'll be redirected to the main pcluster console screen where we can view update progress.

Reviewer: ```suggestion
Once the update starts you will be redirected to the main pcluster manager console screen where you can view update progress.
```

          DisableSimultaneousMultithreading: true
          Efa:
            Enabled: true
            GdrSupport: true
      Networking:
        SubnetIds:
          - subnet-123456789
        PlacementGroup:
          Enabled: true

>       Image:

Reviewer: Not needed. Please remove.
